TokenRateTest.php

```diff
@@ -0,0 +1,50 @@
+<?
+
+namespace iFixit\TokenBucket;
+
+error_reporting(E_ALL);
+
+use \DateInterval;
+
+use PHPUnit_Framework_TestCase;
+
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'TokenRate.php';
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'TokenBucket.php';
+
+
+class TokenRateTest extends PHPUnit_Framework_TestCase {
+   public function testRate() {
+      $rate = new TokenRate(1, new DateInterval("PT01S"));
+      $this->assertSame(1.0, $rate->getRate());
+   }
+
+   public function testLargerRate() {
+      $rate = new TokenRate(10, new DateInterval("PT10S"));
+      $this->assertSame(1.0, $rate->getRate());
+   }
+
+   public function testZeroRate() {
+      $rate = new TokenRate(0, new DateInterval("PT10S"));
+      $this->assertSame(0, $rate->getRate());
+   }
+
+   /**
+    * I'm not really sure what this means for token buckets, but it's still a
+    * rate.
+    */
+   public function testNegativeTokenRate() {
+      $rate = new TokenRate(-5, new DateInterval("PT10S"));
```

Comment:
I half-agree. I think that using DateInterval is the correct approach for timing (@danielbeardsley, it does in fact support microtime).

But i think that the usage here could use some more explanation. That "PT10S" is pretty opaque. Maybe some thorough comments on the TokenRate class @MooreGuy?